tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/DotnetTest/CoverageGetCoverageResultIntegration.cs

```diff
@@ -0,0 +1,87 @@
+// <copyright file="CoverageGetCoverageResultIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+using System;
+using System.ComponentModel;
+using Datadog.Trace.Ci.Ipc;
+using Datadog.Trace.Ci.Ipc.Messages;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Utilities;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.DotnetTest;
+
+#pragma warning disable SA1201
+
+/// <summary>
+/// Coverlet.Core.CoverageResult Coverlet.Core.Coverage::GetCoverageResult() calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "coverlet.core",
+    TypeName = "Coverlet.Core.Coverage",
+    MethodName = "GetCoverageResult",
+    ReturnTypeName = "Coverlet.Core.CoverageResult",
+    ParameterTypeNames = [],
+    MinimumVersion = "6.0.0",
+    MaximumVersion = "6.*.*",
+    IntegrationName = DotnetCommon.DotnetTestIntegrationName)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public class CoverageGetCoverageResultIntegration
+{
+    internal static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception? exception, in CallTargetState state)
+        where TReturn : ICoverageResultProxy
+    {
+        if (DotnetCommon.IsDataCollectorDomain &&
+            instance?.GetType().Assembly() is { } assembly &&
+            assembly.GetType("Coverlet.Core.CoverageSummary") is { } coverageSummaryType)
+        {
+            var coverageSummary = Activator.CreateInstance(coverageSummaryType).DuckCast<ICoverageSummaryProxy>();
+            var coverageDetails = coverageSummary!.CalculateLineCoverage(returnValue.Modules);
+            var percentage = coverageDetails.Percent;
+            DotnetCommon.Log.Information("CoverageGetCoverageResult.Percentage: {Value}", percentage);
+
+            // Extract session variables (from out of process sessions)
+            if (SpanContextPropagator.Instance.Extract(
+                    EnvironmentHelpers.GetEnvironmentVariables(),
+                    new DictionaryGetterAndSetter(DictionaryGetterAndSetter.EnvironmentVariableKeyProcessor)) is { } sessionContext)
+            {
+                try
+                {
+                    var name = $"session_{sessionContext.SpanId}";
```

Comment:
is `sessionContext.SpanId` the same as the `SessionId` values used elsewhere?

Reply:
yup!